src/sonic-host-services/scripts/hostcfgd

```diff
@@ -758,136 +783,124 @@ class KdumpCfg(object):
-            return
+        if not self.ntp_servers:
+            syslog.syslog(syslog.LOG_INFO, "No ntp server when global config change, do nothing")
+            return 
 
-        if (new_src != orig_src):
+        if orig_src_set != new_src_set:
             syslog.syslog(syslog.LOG_INFO, "ntp global update for source intf old {} new {}, restarting ntp-config"
-                          .format(orig_src, new_src))
+                          .format(orig_src_set, new_src_set))
             cmd = 'systemctl restart ntp-config'
             run_cmd(cmd)
-        else:
-            if (new_vrf != orig_vrf):
-                syslog.syslog(syslog.LOG_INFO, "ntp global update for vrf old {} new {}, restarting ntp service"
-                              .format(orig_vrf, new_vrf))
-                cmd = 'service ntp restart'
-                run_cmd(cmd)
-
-    def ntp_server_update(self, key, data, isLoad):
-        syslog.syslog(syslog.LOG_INFO, 'ntp server update key {} data {}'.format(key, data))
-
-        # during load, restart ntp-config regardless if ntp server is configured or not
-        if isLoad == True:
-            if data != {}:
-                self.has_ntp_servers = True
-        else:
-            # for runtime ntp server change, to determine if there is ntp server configured, need to
-            # get from configDB, as delete triggers 2 event handling
-            ntp_servers_tbl = self.config_db.get_table('NTP_SERVER')
-            if ntp_servers_tbl != {}:
-                self.has_ntp_servers = True
-            else:
-                self.has_ntp_servers = False
+        elif new_vrf != orig_vrf:
+            syslog.syslog(syslog.LOG_INFO, "ntp global update for vrf old {} new {}, restarting ntp service"
+                            .format(orig_vrf, new_vrf))
+            cmd = 'service ntp restart'
+            run_cmd(cmd)
 
-        cmd = 'systemctl restart ntp-config'
-        syslog.syslog(syslog.LOG_INFO, 'ntp server update, restarting ntp-config, ntp server exists {}'.format(self.has_ntp_servers))
+    def ntp_server_update(self, key, op):
+        syslog.syslog(syslog.LOG_INFO, 'ntp server update key {}'.format(key))
 
-        run_cmd(cmd)
+        restart_config = False
+        if op == "SET" and key not in self.ntp_servers:
+            restart_config = True
+            self.ntp_servers.add(key)
+        elif op == "DEL" and key in self.ntp_servers:
+            restart_config = True
+            self.ntp_servers.remove(key)
 
+        if restart_config:
+            cmd = 'systemctl restart ntp-config'
+            syslog.syslog(syslog.LOG_INFO, 'ntp server update, restarting ntp-config, ntp servers configured {}'.format(self.ntp_servers))
+            run_cmd(cmd)
 
 class HostConfigDaemon:
     def __init__(self):
+        # Just a sanity check to verify if the CONFIG_DB has been initialized
+        # before moving forward
```

Comment:
Why do we need this explicitelly?

Reply:
This was there before, just retained it.

Reply:
I wonder if we can remove it, those daemons that use swsscommon.SubscriberStateTable don't use ConfigDBConnector's wait_for_init.

Reply:
This was probably added because this script used to start very early, and the implementers wanted to wait until the CONFIG_DB is initialized/redis is up.   Although, you may argue that this can be removed. But since it doesn't affect anything, i think it's better to have as a sanity check.  

https://github.com/Azure/sonic-swss-common/blob/master/common/configdb.cpp#L28